src/plugins/home/public/application/components/guided_onboarding/getting_started.tsx

```diff
@@ -98,10 +115,66 @@ export const GettingStarted = () => {
 
   const isDarkTheme = uiSettings.get<boolean>('theme:darkMode');
   const activateGuide = async (useCase: UseCase, guideState?: GuideState) => {
-    await guidedOnboardingService?.activateGuide(useCase as GuideId, guideState);
-    // TODO error handling https://github.com/elastic/kibana/issues/139798
+    try {
+      await guidedOnboardingService?.activateGuide(useCase as GuideId, guideState);
+    } catch (err) {
+      getServices().toastNotifications.addDanger({
+        title: i18n.translate('home.guidedOnboarding.gettingStarted.activateGuide.errorMessage', {
+          defaultMessage: 'Unable to start the guide. Please try again later.',
+        }),
+        text: err.message,
+      });
+    }
   };
 
+  if (isLoading) {
+    return (
+      <KibanaPageTemplate.EmptyPrompt
+        title={<EuiLoadingSpinner size="xl" />}
+        body={
+          <EuiText color="subdued">
+            {i18n.translate('home.guidedOnboarding.gettingStarted.loadingIndicator', {
+              defaultMessage: 'Loading setup guide state...',
+            })}
+          </EuiText>
+        }
+        data-test-subj="onboarding--loadingIndicator"
+      />
+    );
+  }
+
+  if (isError) {
+    return (
+      <KibanaPageTemplate.EmptyPrompt
+        iconType="alert"
+        color="danger"
+        title={
+          <h2>
+            {i18n.translate('home.guidedOnboarding.gettingStarted.errorSectionTitle', {
+              defaultMessage: 'Unable to load guide state',
```

Comment:
I'll defer to @kellyemurphy , but I wonder if we should make the error (and loading copy above) more generic here. A user probably doesn't understand the concept of "guide state". Maybe something like "Unable to load guides"